ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/events/TransactionsWriter.scala

```diff
@@ -133,40 +168,35 @@ private[dao] final class TransactionsWriter(
         witnesses = disclosureForTransactionTree,
       )
 
-    eventBatches.foreach(_.execute())
-    flatTransactionWitnessesBatch.foreach(_.execute())
-    complementWitnessesBatch.foreach(_.execute())
-
     val contractBatches = contractsTable.prepareBatchInsert(
       ledgerEffectiveTime = ledgerEffectiveTime,
       transaction = transaction,
       divulgedContracts = divulgedContracts,
     )
 
-    for ((deleted, deleteContractsBatch) <- contractBatches.deletions) {
-      val deleteWitnessesBatch = contractWitnessesTable.prepareBatchDelete(deleted.toSeq)
-      assert(deleteWitnessesBatch.nonEmpty, "No witness found for contracts marked for deletion")
-      // Delete the witnesses first to respect the foreign key constraint of the underlying storage
-      deleteWitnessesBatch.get.execute()
-      deleteContractsBatch.execute()
-    }
-
-    for ((_, insertContractsBatch) <- contractBatches.insertions) {
-      insertContractsBatch.execute()
-    }
+    val deleteWitnessesBatch =
+      for ((deleted, _) <- contractBatches.deletions) yield {
+        val deletedWitnessesBatch = contractWitnessesTable.prepareBatchDelete(deleted.toSeq)
+        assert(deletedWitnessesBatch.nonEmpty, "No witness found for contracts marked for deletion")
+        deletedWitnessesBatch.get
```

Comment:
I know you didn't introduce this here, but: This feels like a `getOrElse { throw … }` would be cleaner.

Reply:
Yeah, maybe a blunter approach was more readable before but now it's way better to use `getOrElse`. Good point, thanks!

Reply:
Addressed by https://github.com/digital-asset/daml/pull/5930/commits/d184b38da54f98383c779310e3d747b68143a224